### PR TITLE
Add slash command example to quickstart

### PR DESCRIPTION
**Pull request by HyperGH**
## Summary

Currently the quickstart is written in a way that is best-suited for the older commands framework. This PR aims to somewhat modernize the quickstart and introduce users to slash commands earlier on.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: docs/quickstart.rst [hunk 1]
```diff
@@ -77,3 +76,42 @@ On other systems:
...
+
+.. code-block:: python3
+
+    import discord
+
+    bot = discord.Bot()
+
+    @bot.event
+    async def on_ready():
+        print(f'We have logged in as {bot.user}')
```
Comment (krittick) [inline docs/quickstart.rst:95]:
```suggestion
        print(f"We have logged in as {bot.user}")
```

File: docs/quickstart.rst [hunk 2]
```diff
@@ -77,3 +76,42 @@ On other systems:
...
+
+    @bot.slash_command(name="hello", guild_ids=[your, guild_ids, here])
+    async def hello_slash(ctx):
+        await ctx.respond("Hello!")
+
+    bot.run('your token here')
```
Comment (krittick) [inline docs/quickstart.rst:101]:
```suggestion
    bot.run("your token here")
```

**Comment by krittick** [comment]
I feel like completely replacing the existing `discord.Client`-based example instead of explaining the differences would be better and easier to understand for new users. Otherwise I like where this is headed.

**Comment by HyperGH** [comment]
> I feel like completely replacing the existing `discord.Client`-based example instead of explaining the differences would be better and easier to understand for new users. Otherwise I like where this is headed.

I do believe that showing off the basic functionality without slash commands still merits some value, as not everyone wants to use them, and not everyone necessarily even wants their bot to have commands in the first place.

File: docs/quickstart.rst [hunk 3]
```diff
@@ -77,3 +76,42 @@ On other systems:
     $ python3 example_bot.py
 
 Now you can try playing around with your basic bot.
+
+A Minimal Bot with Slash Commands
+---------------
```
Comment (VincentRPS) [inline docs/quickstart.rst:81]: Why is the line shorter than the text?

File: docs/quickstart.rst [hunk 4]
```diff
@@ -77,3 +76,42 @@ On other systems:
...
+
+    bot.run("your token here")
+
+Let's look at the differences compared to the previous example, step-by-step:
+
+1. The first line remains unchanged.
+2. Next, we create an instance of :class:`Bot`. This is different from :class:`Client`, as it supports
+   slash command creation and other features, while inheriting all the features of :class:`Client`.
+3. We then use the :meth:`Bot.slash_command` decorator to register a new slash command. The first
+   keyword-argument, ``name``, sets the command's name to ``hello``. The ``guild_ids`` attribute contains
```
Comment (VincentRPS) [inline docs/quickstart.rst:109]: This makes name sound like a required argument

File: docs/quickstart.rst [hunk 5]
```diff
@@ -55,8 +55,7 @@ There's a lot going on here, so let's walk you through it step by step.
    sure that we ignore messages from ourselves. We do this by checking if the :attr:`Message.author`
    is the same as the :attr:`Client.user`.
 5. Afterwards, we check if the :class:`Message.content` starts with ``'$hello'``. If it does,
-   then we send a message in the channel it was used in with ``'Hello!'``. This is a basic way of 
-   handling commands, which can be later automated with the :doc:`./ext/commands/index` framework.
+   then we send a message in the channel it was used in with ``'Hello!'``.
```
Comment (VincentRPS) [inline docs/quickstart.rst:58]: Why did you remove stuff from this?

File: docs/quickstart.rst [hunk 6]
```diff
@@ -40,7 +40,7 @@ It looks something like this:
 Let's name this file ``example_bot.py``. Make sure not to name it ``discord.py`` as that'll conflict
 with the library.
 
-There's a lot going on here, so let's walk you through it step by step.
+There's a lot going on here, so let's walk you through it step by step:
```
Comment (VincentRPS) [inline docs/quickstart.rst:43]: This didn’t need to be changed imo

File: docs/quickstart.rst [hunk 7]
```diff
@@ -77,3 +76,42 @@ On other systems:
...
+
+As a continuation, let's create a bot that registers a simple slash command!
+
+It looks something like this:
+
+.. code-block:: python3
+
+    import discord
+
+    bot = discord.Bot()
```
Comment (VincentRPS) [inline docs/quickstart.rst:91]: Basically what @krittick said.

**Review by VincentRPS — CHANGES_REQUESTED** [review]
There are some weird changes not in sync to what this pr should do.

File: docs/quickstart.rst (hunk 5 above)
Comment (HyperGH) [reply docs/quickstart.rst:58]: Removed it in order to not direct users towards ext.commands when there is a slash example 2 lines below it.

File: docs/quickstart.rst (hunk 6 above)
Comment (HyperGH) [reply docs/quickstart.rst:43]: I thought it makes more sense, but I'm fine with both.

File: docs/quickstart.rst (hunk 5 above)
Comment (VincentRPS) [reply docs/quickstart.rst:58]: But ext.commands work fully well with slash commands

Comment (HyperGH) [reply docs/quickstart.rst:58]: It does, but isn't what I would consider the logical next step after completing that example, as this text would imply.

File: docs/quickstart.rst (hunk 3 above)
Comment (HyperGH) [reply docs/quickstart.rst]: Fixed

File: docs/quickstart.rst (hunk 4 above)
Comment (HyperGH) [reply docs/quickstart.rst:109]: I think I'll actually just remove name entirely to simplify this example.

File: docs/quickstart.rst (hunk 5 above)
Comment (VincentRPS) [reply docs/quickstart.rst:58]: I mean prefixed commands have there upside and iirc discord.Bot is still missing some ext.commands features, I am also planning on deprecating discord.Bot soon and making prefix optional in commands.Bot

Comment (HyperGH) [reply docs/quickstart.rst:58]: At this point we are getting into library design talk, and as I cannot really see into the future of how the library will shape up, I don't think I can just the example to that as of now. All of this can be adjusted as the library further develops :)

Comment (VincentRPS) [reply docs/quickstart.rst:58]: Ye I just think this shouldn’t be changed but I would like others opinion

Comment (Lulalaby) [reply docs/quickstart.rst:58]: Please revert this change

**Review by VincentRPS — APPROVED** [review]

File: docs/quickstart.rst [hunk 8]
```diff
@@ -77,3 +77,41 @@ On other systems:
...
+        await ctx.respond("Hello!")
+
+    bot.run("your token here")
+
+Let's look at the differences compared to the previous example, step-by-step:
+
+1. The first line remains unchanged.
+2. Next, we create an instance of :class:`Bot`. This is different from :class:`Client`, as it supports
+   slash command creation and other features, while inheriting all the features of :class:`Client`.
+3. We then use the :meth:`Bot.slash_command` decorator to register a new slash command.
```
Comment (Dorukyum) [inline docs/quickstart.rst:109]:
```suggestion
2. Next, we create an instance of :class:`.Bot`. This is different from :class:`.Client`, as it supports
   slash command creation and other features, while inheriting all the features of :class:`.Client`.
3. We then use the :meth:`.Bot.slash_command` decorator to register a new slash command. 
```
I think these are the correct references.

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]
